src/ParseObject.js

```diff
@@ -1821,15 +1834,19 @@ class ParseObject {
    *
    * @param {String} name Name of Pin.
    * @param {Array} objects A list of <code>Parse.Object</code>.
+   * @return {Promise} A promise that is fulfilled when the pin completes.
    * @static
    */
-  static async pinAllWithName(name: string, objects: Array<ParseObject>): Promise {
+  static pinAllWithName(name: string, objects: Array<ParseObject>): Promise {
     const localDatastore = CoreManager.getLocalDatastore();
-    if (localDatastore.checkIfEnabled()) {
-      for (const object of objects) {
-        await localDatastore._handlePinWithName(name, object);
-      }
+    if (!localDatastore.isEnabled) {
+      return Promise.reject('Parse.enableLocalDatastore() must be called first');
+    }
+    let promise = Promise.resolve();
+    for (const object of objects) {
+      promise = promise.then(() => localDatastore._handlePinWithName(name, object));
```

Comment:
this doesn't seem ok.  I'd think you'd want:
```
const promises = objects.map(object =>  localDatastore._handlePinWithName(name, object);
return Promise.all(promises);
```

as you have it you just keep writing over promise.  Also not sure why you'd resolve a promise to just promise.then() and add a new promise....or i'm missing something...

Reply:
I tried that but data was getting overridden. I can try a better implementation

Reply:
or 

```
const promises = objects.map(localDatastore._handlePinWithName.bind(null, name));
return Promise.all(promises);
```

:).

Reply:
I changed it to handle all pins instead of them individually, this allows batching the objects since they all have the same name pin.